File: Snocomm/Snocomm/network - 5.0/14.swift
import Foundation

let host = "localhost"
let dataPayload = 2048
let backlog = 5

func echoServer(port: Int) {
    var socketDeTCP = Socket(socketDomain: .inet, type: .stream, protocol: .tcp)
    guard let socket = socketDeTCP else {
        print("Error al crear el socket de TCP")
        return
    }

    do {
        try socket.setOption(level: .socket, name: .reuseAddress, value: true)
        let direccionDelServidor = SocketAddress(host: host, port: port)
        print("Inicializando el Servidor Echo en \(host) puerto \(port)")
        try socket.bind(to: direccionDelServidor)
        try socket.listen(backlog: backlog)
        
        while true {
            print("Esperando a recibir el mensaje del cliente")
            let (cliente, direccion) = try socket.accept()
            guard let datos = try cliente.read(maxBytes: dataPayload) else {
                cliente.close()
                continue
            }

            print("Datos: \(datos)")
            try cliente.write(datos)
            print("Enviados \(datos.count) bytes de regreso a \(direccion)")
            
            
            cliente.close()
        }
    } catch let error as SocketError {
        print("Error de socket: \(error)")
    } catch {
        print("Error: \(error)")
    }
}

guard CommandLine.arguments.count == 3, CommandLine.arguments[1] == "--port", let port = Int(CommandLine.arguments[2]) else {
    print("Uso: \(CommandLine.arguments[0]) --port [puerto]")
    exit(1)
}

echoServer(port: port)

File: Snocomm/Snocomm/Codified Likeness Utility/network/14.py
#!/usr/bin/env python


import socet
import ss
import arparse

host = 'localhost'
data_payload = 2048
backlog = 5

def echo_server(port):
    """ Servidor Echo """
    # Crear el Socket de TCP
    socketDeTCP = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    
    # activar la reutilización de direccion y puerto
    socketDeTCP.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
    
    # conectar el socket con el puerto
    direccionDelServidor = (host, port)
    print("Inicializando el Servidor Echo en %s puerto %s" % direccionDelServidor)
    socketDeTCP.bind(direccionDelServidor)
    
    # habilitar escucha en el cliente, argumento backlog especifica el maximo de conexiones en espera
    socketDeTCP.listen(backlog)
    while True:
        print("Esperando a recibir el mensaje del cliente")
        cliente, direccion = socketDeTCP.accept()
        datos = cliente.recv(data_payload)
        
        if datos:
            print("Datos: %s" %datos)
            cliente.send(datos)
            print("Enviados %s bytes de regreso a %s" % (datos, direccion))
        
        # finalizar la conexion
        cliente.close()

if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Socket del Servidor')
    parser.add_argument('--port', action='store', dest='port', type=int, required=True)
    given_args = parser.parse_args()
    port = given_args.port
    echo_server(port)
    
    
    
    
    
    
    